d8bd2b67c4f2df09348c0e68df2d5ac4bb65d50f
src/main/java/org/apache/platypus/server/grpc/LuceneServer.java
--- 
+++ 
@@ -365,7 +365,10 @@
 
                 private void finishIndexingJob() throws IOException {
                     for (String indexName : globalState.getIndexNames()) {
-                        globalState.getIndex(indexName).getShard(0).maybeRefreshBlocking();
+                        ShardState shard = globalState.getIndex(indexName).getShard(0);
+                        if (shard.isStarted()) {
+                            shard.maybeRefreshBlocking();
+                        }
                     }
                 }
             };